keyboard.c

```diff
@@ -327,17 +360,29 @@ kbd_print_key_stdout(struct kbd *kb, struct key *k) {
 }
 
 void
-kbd_draw_key(struct kbd *kb, struct key *k, bool pressed) {
+kbd_draw_key(struct kbd *kb, struct key *k, bool pressed, bool swiped) {
 	struct drwsurf *d = kb->surf;
 	const char *label = (kb->mods & Shift) ? k->shift_label : k->label;
 	if (kb->debug)
 		fprintf(stderr, "Draw key +%d+%d %dx%d -> %s\n", k->x, k->y, k->w, k->h,
 		        label);
 	struct clr_scheme *scheme = (k->scheme == 0) ? &(kb->scheme) : &(kb->scheme1);
-	Color *fill = pressed ? &scheme->high : &scheme->fg;
-	draw_inset(d, k->x, k->y, k->w, k->h, KBD_KEY_BORDER, *fill);
+	if (swiped) {
+		Color *fill = &scheme->swipe;
+		draw_over_inset(d, k->x, k->y, k->w, k->h, KBD_KEY_BORDER, *fill);
+	} else {
+		Color *fill = pressed ? &scheme->high : &scheme->fg;
+		draw_inset(d, k->x, k->y, k->w, k->h, KBD_KEY_BORDER, *fill);
+	}
 	drw_draw_text(d, scheme->text, k->x, k->y, k->w, k->h, label);
 }
+void
+kbd_draw_press(struct kbd *kb, struct key *k, bool pressed) {
+	kbd_draw_key(kb, k, pressed, false);
+}
+void kbd_draw_swipe(struct kbd *kb, struct key *k) {
+	kbd_draw_key(kb, k, false, true);
+}
```

Comment:
Is there a good reason not to inline these? The change in arguments looks very trivial. I know this means essentially nothing to the compiler, but I think it makes the code a little less readable.

Reply:
I was probably motivaned by "DRY" or keeping the patch small, but ~8 repeated lines doesn't seem too bad...

kbd_draw_key in this patch (with 4 arguments) doesn't seem particularly useful, though semantically, I think 3 names make sense, maybe an implementation using enums would be better for readability and extensibility, e.g.

```c
kbd_draw_key(kb, k, KEY_PRESS);
kbd_draw_key(kb, k, KEY_UNPRESS);
kbd_draw_key(kb, k, KEY_SWIPE);
```

Reply:
I think the approach you described just now is the best of the three options, if you're motivated to do that I approve. Or I could take the patch as-is and do it later, but I'd be grateful if you tried it :smiley:

Reply:
@jjsullivan5196  I made the change